### PR TITLE
45751 Expand individual nav warning

### PR DESCRIPTION
**Pull request by vipin14904**
WIP. Automation testing is in progress.

### CODE REVIEW

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs [hunk 1]
```diff
@@ -51,7 +53,7 @@ public void WhenAddRadioNavigationWarningsReturnTrueInRequest_ThenNewRecordIsCre
         {
             var httpContext = new DefaultHttpContext();
             var tempData = new TempDataDictionary(httpContext, A.Fake<ITempDataProvider>());
```
Comment (usha13529) [inline Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs:55]: Remove unused variable tempData  and httpContext.

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs [hunk 2]
```diff
@@ -60,12 +62,44 @@ public void WhenAddRadioNavigationWarningsReturnTrueInRequest_ThenNewRecordIsCre
...
 
+
         [Test]
-        public void WhenICallIndexViewWithParameters_ThenReturnView()
+        public void WhenAddRadioNavigationWarningsReturnFalseInRequest_ThenNewRecordIsNotCreated()
         {
-            A.CallTo(() => _fakeRnwService.GetRadioNavigationWarningsForAdmin(1, 0, null, string.Empty)).Returns(GetFakeRadioNavigationWarningsForAdmin());
-            Task<IActionResult> result = _controller.Index(pageIndex: 1, warningType: 1, year: 2020);
+            var httpContext = new DefaultHttpContext();
+            var tempData = new TempDataDictionary(httpContext, A.Fake<ITempDataProvider>());
```
Comment (usha13529) [inline Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs:70]: Remove unused variable tempData  and httpContext.

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs [hunk 3]
```diff
@@ -60,12 +62,44 @@ public void WhenAddRadioNavigationWarningsReturnTrueInRequest_ThenNewRecordIsCre
...
+
             Assert.IsInstanceOf<Task<IActionResult>>(result);
+            Assert.IsNull(_controller.TempData["message"]);
+        }
+
+        [Test]
+        public void WhenAddRadioNavigationWarningsWithInValidModel_ThenNewRecordIsNotCreated()
+        {
+            var httpContext = new DefaultHttpContext();
+            var tempData = new TempDataDictionary(httpContext, A.Fake<ITempDataProvider>());
```
Comment (usha13529) [inline Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs:84]: Remove unused variable tempData  and httpContext.

**Review by usha13529 — CHANGES_REQUESTED** [review]
Minor review comments to fix please.

**Review by nevillejrbrown — APPROVED** [review]

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs (hunk 1 above)
Comment (ganpat-tech) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs]: Thanks for highlighting done required changes.

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs (hunk 2 above)
Comment (ganpat-tech) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs]: done required changes.

File: Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs (hunk 3 above)
Comment (ganpat-tech) [reply Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/RadioNavigationalWarningsAdminControllerTest.cs]: done required changes.

**Review by usha13529 — APPROVED** [review]
Thanks for the changes @ganpat-tech. Looks good to me.

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Models/RadioNavigationalWarning/RadioNavigationalWarningsData.cs [hunk 4]
```diff
@@ -12,5 +12,8 @@ public class RadioNavigationalWarningsData
 
         [DisplayName("Date Time Group")]
         public string DateTimeGroupRnwFormat { get; set; }
+
+        public string Content { get; set; }
+        public string WarningType { get; set; }
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Models/RadioNavigationalWarning/RadioNavigationalWarningsData.cs:18]: Thanks For highlighting, done required changes

File: UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/RadioNavigationalWarningsAdminController.cs [hunk 5]
```diff
@@ -44,7 +44,7 @@ public async Task<IActionResult> Create(RadioNavigationalWarning radioNavigation
                     TempData["message"] = "Record created successfully!";
                     _logger.LogInformation(EventIds.CreateNewRNWRecordCompleted.ToEventId(), "Maritime safety information create new RNW record request completed for _X-Correlation-ID:{correlationId}", GetCurrentCorrelationId());
 
-                    return RedirectToAction(nameof(Index), new { reLoadData = true });
+                    return RedirectToAction(nameof(Index));
```
[the remark this answers is not in the document]
Comment (ganpat-tech) [reply UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Controllers/RadioNavigationalWarningsAdminController.cs:47]: changes already in place.

**Review by barrie-cooper — APPROVED** [review]
LGTM, thanks @ganpat-tech

**Pull request merged**